### PR TITLE
Item history

### PR DESCRIPTION
**Pull request by flange-ipb**
- more history navigation tests for ItemState and HistoryOperation; also discoved and fixed missing boundary tests when moving back and forth in the history via HistoryOperation
- add getters to be used in the UI to reflect the before-first/last position in the history list (copy of ItemBean.isPreviousVersionButtonDisabled() and ItemBean.isNextVersionButtonDisabled(), which will be removed in the following UI polish)

### CODE REVIEW

File: ui/src/main/java/de/ipb_halle/lbac/items/bean/ItemState.java [hunk 1]
```diff
@@ -119,8 +119,25 @@ public Date getFollowingKey(Date d) {
...
+     */
+    public boolean isLastHistoryItem() {
+        return currentHistoryDate == null;
+    }
+
+    /**
+     * @return true if the currently selected item version is the first revision.
+     */
     public boolean isStartingHistoryItem() {
+        // TODO: does not work this way
```
Comment (FabianMauz) [inline ui/src/main/java/de/ipb_halle/lbac/items/bean/ItemState.java]: Is the comment obselete or is it still an issue?

File: ui/src/main/java/de/ipb_halle/lbac/items/bean/ItemState.java [hunk 2]
```diff
@@ -119,8 +119,25 @@ public Date getFollowingKey(Date d) {
...
         return getPreviousKey(currentHistoryDate) == null;
     }
+
+    /**
+     * @return date when this revision was done
+     */
+    public Date getChangeDate() {
+        return getPreviousKey(currentHistoryDate);
```
Comment (FabianMauz) [inline ui/src/main/java/de/ipb_halle/lbac/items/bean/ItemState.java:149]: What happens, if the currentHistoryDate is the first date in the historyKeySet?

**Review by FabianMauz — APPROVED** [review]
